frontend/model/captureLogs.js

```diff
@@ -0,0 +1,176 @@
+  localStorage.setItem('giConsole/count', entriesCount)
+
+  verifyLogsSize()
+
+  // To avoid infinite loop because we log all selector calls, we run sbp calls
+  // here in a roundabout way by getting the function to which they're mapped.
+  // The reason this works is because the entire `sbp` domain is blacklisted
+  // from being logged in main.js.
+  sbp('sbp/selectors/fn', 'okTurtles.events/emit')(CAPTURED_LOGS, lastEntry)
+}
+
+function verifyLogsSize () {
+  if (entriesCount % ENTRIES_MARKER === 0) {
+    const markers = getMarkers()
+    // Save entry as a marker to be later deleted when logs are too big.
+    markers.push(lastEntry)
+    localStorage.setItem('giConsole/markers', JSON.stringify(markers))
+  }
+
+  if (entriesCount >= ENTRIES_LIMIT) {
+    const markers = getMarkers()
+    let toDelete = ENTRIES_MARKER // delete the latest x entries
+    let prevEntry = markers.shift()
+    do {
+      const entry = JSON.parse(localStorage.getItem(`giConsole/${prevEntry}`)) || {}
+      localStorage.removeItem(`giConsole/${prevEntry}`)
+      prevEntry = entry.prev
+      entriesCount--
+      toDelete--
+    } while (toDelete && prevEntry)
+    localStorage.setItem('giConsole/markers', JSON.stringify(markers))
+    localStorage.setItem('giConsole/count', entriesCount)
+  }
+}
+
+export function captureLogsStart () {
+  console.debug('captureLogsStart called')
+  isCapturing = true
+  lastEntry = localStorage.getItem('giConsole/lastEntry')
+  entriesCount = +localStorage.getItem('giConsole/count') || 0
+
+  // Set a new visit or session - useful to understand logs through time.
+  // NEW_SESSION -> The user opened a new browser or tab.
+  // NEW_VISIT -> The user comes from an ongoing session (refresh or login)
+  const isNewSession = !sessionStorage.getItem('NEW_SESSION')
+  if (isNewSession) { sessionStorage.setItem('NEW_SESSION', 1) }
+  captureLog(isNewSession ? 'NEW_SESSION' : 'NEW_VISIT')
+
+  // Subscribe to appLogsFilter
+  const state = sbp('state/vuex/state')
+  appLogsFilter = state.appLogsFilter || []
+  sbp('okTurtles.events/on', SET_APP_LOGS_FILTER, filter => { appLogsFilter = filter })
+
+  console.debug('Starting to capture logs of type:', appLogsFilter)
+
+  // Override the console to start capturing the logs
+  if (!isConsoleOveridden) {
+    // avoid duplicated captures, in case captureLogsStart
+    // is called multiple times. (ex: login twice in the same visit)
+    isConsoleOveridden = true
+
+    // NOTE: Find a way to capture logs without messing up with log file location.
+    // console.log() doesnt include stack trace, so when logged, we can't access
+    // where the log came from (file name), which difficults debugging if needed.
+    window.console = new Proxy(window.console, {
+      get: (obj, type) => (...args) => {
+        obj[type](...args)
+        if (isCapturing && appLogsFilter.includes(type)) {
+          captureLog(type, ...args)
+        }
+      }
+    })
+  }
+}
```

Comment:
I'm confused about this comment. You recommend to not use `NEW_SESSION` or  `NEW_VISIT` but the code snippet you share still contains those... I think it's valuable information to know if the user did a refresh or visit "a brand new" page. 

> it will speed up the log page view by requiring fewer comparisons.
I think this is a micro-optimization, the comparison will only happen when it doesn't match a "normal" filter. 

```js
    // isEntryNew is only called if `this.form.filter.includes(type)` is falsy. It doesn't happen very often.
    prettyLogs () {
      return this.ephemeral.logs
        .filter(({ type }) => this.form.filter.includes(type) || isEntryNew(type))
        // ...
    }
```

Reply:
> You recommend to not use NEW_SESSION or NEW_VISIT but the code snippet you share still contains those

It doesn't contain them. It contains `console.info`, whereas before you had `console.NEW_SESSION` (effectively).

Reply:
Oh, gotcha! Fixed.